[nvidia] The build with latest DPC++ open source compiler is broken

# Environment
The latest open source compiler build with cuda support (but the failures are tracked down to the one of builds in early September).

# Steps to reproduce
```
mkdir -p build
cd build
cmake .. -DCMAKE_BUILD_TYPE=release (or debug) -DDNNL_CPU_RUNTIME=DPCPP (or NONE) -DDNNL_GPU_RUNTIME=DPCPP -DDNNL_GPU_VENDOR=NVIDIA -DONEDNN_BUILD_GRAPH=OFF
cmake --build . --target dnnl
```

# Observed behavior
```
[311/630] Building CXX object src/gpu/sycl/CMakeFiles/dnnl_gpu_sycl.dir/ref_eltwise.cpp.o
FAILED: src/gpu/sycl/CMakeFiles/dnnl_gpu_sycl.dir/ref_eltwise.cpp.o
/compiler/cuda/compiler/bin/clang++ -DCL_TARGET_OPENCL_VERSION=300 -DDNNL_DLL -DDNNL_DLL_EXPORTS -DDNNL_ENABLE_CPU_ISA_HINTS -DDNNL_ENABLE_MAX_CPU_ISA -DDNNL_X64=1 -DNGEN_CPP11 -DNGEN_NEO_INTERFACE -DNGEN_NO_OP_NAMES -DNGEN_SAFE -DNGEN_WINDOWS_COMPAT -DTBB_PREVIEW_TASK_ARENA_CONSTRAINTS_EXTENSION=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/build/include -I/include -I/compiler/cuda/tbb/include -I/src -Wno-c++20-extensions -Wno-unknown-cuda-version -idirafter/usr/local/cuda/include -Wno-#pragma-messages -fsycl -fvisibility-inlines-hidden  -ffp-model=precise -fno-reciprocal-math -Wno-unknown-warning-option -Wall -Wno-unknown-pragmas -Werror -fvisibility=internal -march=native  -fPIC -Wformat -Wformat-security -fstack-protector-all   -Wno-pass-failed -Wno-tautological-compare -Wno-pass-failed -Wno-recommended-option -Wno-pass-failed  -fsycl-targets=nvptx64-nvidia-cuda -Wno-linker-warnings -O3 -DNDEBUG -D_FORTIFY_SOURCE=2 -MD -MT src/gpu/sycl/CMakeFiles/dnnl_gpu_sycl.dir/ref_eltwise.cpp.o -MF src/gpu/sycl/CMakeFiles/dnnl_gpu_sycl.dir/ref_eltwise.cpp.o.d -o src/gpu/sycl/CMakeFiles/dnnl_gpu_sycl.dir/ref_eltwise.cpp.o -c /src/gpu/sycl/ref_eltwise.cpp
In file included from ../src/gpu/sycl/ref_eltwise.cpp:17:
In file included from ../src/gpu/sycl/ref_eltwise.hpp:22:
In file included from ../src/gpu/sycl/sycl_post_ops.hpp:22:
../src/gpu/sycl/sycl_math_utils.hpp:97:23: error: no matching function for call to 'tanh'
   97 |     float l = (float)(::sycl::tanh<float>((float)e));
      |                       ^~~~~~~~~~~~~~~~~~~
/compiler/cuda/compiler/bin/../include/sycl/builtins_vector_gen.hpp:421:195: note: candidate template ignored: requirement 'detail::is_vec_or_swizzle_v<float>' was not satisfied [with T0 = float]
  421 | std::enable_if_t<(detail::is_vec_or_swizzle_v<T0> && detail::is_valid_elem_type_v<T0, float, double, half> && detail::is_valid_size_v<T0, 16, 1, 2, 3, 4, 8>), detail::simplify_if_swizzle_t<T0>> tanh(T0 a0) __NOEXC {
      |                                                                                                                                                                                                   ^
/compiler/cuda/compiler/bin/../include/sycl/builtins_marray_gen.hpp:860:106: note: candidate template ignored: requirement 'detail::is_marray_v<float>' was not satisfied [with T0 = float]
  860 | std::enable_if_t<(detail::is_marray_v<T0> && detail::is_valid_elem_type_v<T0, float, double, half>), T0> tanh(T0 a0) __NOEXC {
      |                                                                                                          ^
1 error generated.
```

--- a/src/gpu/sycl/sycl_math_utils.hpp
+++ b/src/gpu/sycl/sycl_math_utils.hpp
@@ -94,7 +94,7 @@ inline U logsigmoid_fwd(T s) {
 template <typename T, typename U = rem_ref<T>>
 inline U mish_fwd(T s) {
     float e = soft_relu_fwd<float>(s);
-    float l = (float)(::sycl::tanh<float>((float)e));
+    float l = (float)(::sycl::tanh((float)e));
     float val = s * l;
     return (U)(val);
 }

